lucille-core/src/main/java/com/kmwllc/lucille/stage/ReplacePatterns.java

```diff
@@ -22,7 +22,12 @@
  * <br>
  * - regex (List&lt;String&gt;) : A list regex expression to find matches for. Matches will be extracted and placed in the destination fields.
  * <br>
- * - replacement (String) : The String to replace regex matches with.
+ * - replacement (String, Optional) : The String to replace regex matches with. If null, pattern replacement will only take place if
+ * a replacement_field is specified and found in a document.
+ * <br>
+ * - replacement_field (String, Optional): Specify a field in the document that maps to a String. If non-null, replacements of a
+ * pattern within a document will use the string mapped to the replacement_field, if present. Otherwise, the default
+ * replacement String is used (if it is not null).
```

Comment:
minor: please be explicit about what happens if we try to use the default replacement string but it IS null